geometry_msgs/Pose pose
float64 tool_length
